kyverno/kyverno
# feat: add CleanupPolicy validation code to CleanupPolicyHandler

Signed-off-by: Nikhil Sharma <nikhilsharma230303@gmail.com>

## Explanation
This PR add validation code for validating `CleanupPolicies` to `CleanupPolicyHandler`.

Validations added:
1. Check whether the policy name is valid or not.
2. Check whether the `Schedule` specified by the user is in proper cron format or not.
3. Validate the `MatchResources` and `ExcludeResources`.
4. Check if the resultant of match and exclude block is not an empty set.
5. Check whether the CleanupPolicy have permission to delete the matching resources.
